src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -89,7 +91,9 @@ public String toString() {
     public static SamInputResource of(final File file) { return new SamInputResource(new FileInputResource(file)); }
 
     /** Creates a {@link SamInputResource} reading from the provided resource, with no index. */
-    public static SamInputResource of(final Path path) { return new SamInputResource(new PathInputResource(path)); }
```

Comment:
I thought it might be worth it to add an explanation to the PathInputResource constructor in case someone wanders into the code there and wants to know why path inputs have this special thing that the others don't.